packages/middleware/src/integrations/registerIntegrations.ts

```diff
@@ -12,35 +11,55 @@ import type {
   IntegrationsLoaded,
 } from "../types";
 import { defaultErrorHandler } from "../errors/defaultErrorHandler";
+import { LoggerManager, wrapLogger } from "../loggerManager";
 
 export async function registerIntegrations(
```

Comment:
This function is SO unreadable. Please, extract functions, give them names, split operations into block, make it readable.